app/src/main/java/com/discord/simpleast/sample/CustomMarkdownRules.kt

```diff
@@ -54,4 +56,37 @@ object CustomMarkdownRules {
         }
     )
   }
+
+    interface BlockQuoteState<Self: BlockQuoteState<Self>> {
+        var isInQuote: Boolean
+        fun clone(): Self
```

Comment:
Don't think this state should have a clone function. It should be defined by the concrete class, the interface should also only define immutable values.

Reply:
So the issue is that we need to clone the value inside the rule, and once we're there the rule only knows about the interface it has (since that's the only bound on the state's generic type). Therefore we need the interface to know how to clone itself. However I think we could make the values immutable by having this clone method take in a new value for `isInQuote`. I'll go ahead and do that 🙂